cmake/FindCaDiCaL.cmake

```diff
@@ -13,15 +13,80 @@
 # CaDiCaL_INCLUDE_DIR - the CaDiCaL include directory
 # CaDiCaL_LIBRARIES - Libraries needed to use CaDiCaL
 
+include(deps-helper)
+
 find_path(CaDiCaL_INCLUDE_DIR NAMES cadical.hpp)
 find_library(CaDiCaL_LIBRARIES NAMES cadical)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(CaDiCaL
-  DEFAULT_MSG
-  CaDiCaL_INCLUDE_DIR CaDiCaL_LIBRARIES)
+set(CaDiCaL_FOUND_SYSTEM FALSE)
+if(CaDiCaL_INCLUDE_DIR AND CaDiCaL_LIBRARIES)
+    set(CaDiCaL_FOUND_SYSTEM TRUE)
+
+    # Unfortunately it is not part of the headers
+    find_library(CaDiCaL_BINARY NAMES cadical)
+    if(CaDiCaL_BINARY)
+      execute_process(
+        COMMAND ${CaDiCaL_BINARY} --version
+        OUTPUT_VARIALE CaDiCaL_VERSION
+      )
+    else()
+      set(CaDiCaL_VERSION "")
+    endif()
+
+    check_system_version("CaDiCaL")
+endif()
+
+if(NOT CaDiCaL_FOUND_SYSTEM)
+    include(CheckSymbolExists)
+    include(ExternalProject)
+
+    fail_if_include_missing("sys/resource.h" "CaDiCaL")
+
+    # TODO(mpreiner): Use a more recent version?
+    set(CaDiCaL_VERSION "1.2.1")
+
+    # avoid configure script and instantiate the makefile manually
+    # the configure scripts unnecessarily fails for cross compilation
+    # thus we do the bare minimum from the configure script here
+    set(CXXFLAGS "-fPIC -O3 -DNDEBUG -DQUIET -std=c++11")
+    # check for getc_unlocked
+    check_symbol_exists("getc_unlocked" "cstdio" HAVE_UNLOCKED_IO)
+    if(NOT HAVE_UNLOCKED_IO)
+        set(CXXFLAGS "${CXXFLAGS} -DNUNLOCKED")
+    endif()
+
+    ExternalProject_Add(
+        CaDiCaL-EP
+        PREFIX ${DEPS_PREFIX}
+        BUILD_IN_SOURCE ON
+        URL https://github.com/arminbiere/cadical/archive/refs/tags/rel-${CaDiCaL_VERSION}.tar.gz
+        URL_HASH SHA1=9de1176737b74440921ba86395fe5edbb3b131eb
+        CONFIGURE_COMMAND mkdir -p <SOURCE_DIR>/build
+        # avoid configure script, prepare the makefile manually
+        COMMAND sed -e "s,@CXX@,${CMAKE_CXX_COMPILER}," -e "s,@CXXFLAGS@,${CXXFLAGS}," -e "s,@MAKEFLAGS@,-j1," <SOURCE_DIR>/makefile.in > <SOURCE_DIR>/build/makefile
```

Comment:
Indent `COMMAND` since it belongs to `CONFIGURE_COMMAND`?

Reply:
A `COMMAND` generally belongs to the `X_COMMAND` that comes right before it. (see https://cmake.org/cmake/help/latest/module/ExternalProject.html#external-project-definition at "Miscellaneous Options:")

Reply:
I know, that's why I suggested to indent it to make it clearer, but if you prefer it this way I'm fine with it.